File: sample-videochat-webrtc/src/main/java/com/quickblox/sample/groupchatwebrtc/fragments/BaseToolBarFragment.java
package com.quickblox.sample.groupchatwebrtc.fragments;

import android.os.Bundle;
import android.os.Handler;
import android.os.Message;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.quickblox.sample.groupchatwebrtc.R;

import java.lang.ref.WeakReference;

public abstract class BaseToolBarFragment extends Fragment {

    private static final String TAG = BaseToolBarFragment.class.getSimpleName();
    protected Handler mainHandler;
    protected Toolbar toolbar;
    protected ActionBar actionBar;

    abstract int getFragmentLayout();

    @Override
    public void onCreate(Bundle savedInstanceState) {
        setHasOptionsMenu(true);
        mainHandler = new FragmentLifeCycleHandler(this);
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(getFragmentLayout(), container, false);
        initActionBar();
        return view;
    }

    private void initActionBar() {
        toolbar = (Toolbar) getActivity().findViewById(R.id.toolbar_call);

        ((AppCompatActivity) getActivity()).setSupportActionBar(toolbar);
        actionBar = ((AppCompatActivity) getActivity()).getDelegate().getSupportActionBar();
    }


    static class FragmentLifeCycleHandler extends Handler {

        private WeakReference<Fragment> fragmentRef;

        FragmentLifeCycleHandler(Fragment fragment){

            this.fragmentRef = new WeakReference<>(fragment);
        }

        @Override
        public void dispatchMessage(Message msg) {
            Fragment fragment = fragmentRef.get();
            if (fragment == null){
                return;
            }
            if (fragment.isAdded() && (fragment.getActivity() != null)) {
                super.dispatchMessage(msg);
            } else {
                Log.d(TAG, "Fragment under destroying");
            }
        }
    }
}

File: sample-videochat-kotlin/src/main/java/com/quickblox/sample/videochatkotlin/fragments/BaseToolBarFragment.kt
package com.quickblox.sample.videochatkotlin.fragments

import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.os.Handler
import android.os.Message
import android.support.v4.app.Fragment
import android.support.v4.content.ContextCompat
import android.support.v7.app.ActionBar
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.quickblox.chat.QBChatService
import com.quickblox.sample.videochatkotlin.R
import java.lang.ref.WeakReference

abstract class BaseToolBarFragment : Fragment() {
    private var TAG = BaseToolBarFragment::class.java.simpleName
    lateinit var mainHandler: Handler
    lateinit var actionBar: ActionBar

    internal abstract val fragmentLayout: Int

    override fun onCreate(savedInstanceState: Bundle?) {
        setHasOptionsMenu(true)
        mainHandler = FragmentLifeCycleHandler(this)
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(fragmentLayout, container, false)
        actionBar = (activity as AppCompatActivity).delegate.supportActionBar!!
        initActionBar()
        return view
    }

    open fun initActionBar() {
        actionBar.setTitle(String.format(QBChatService.getInstance().user.login))
        actionBar.setBackgroundDrawable(ColorDrawable(ContextCompat.getColor(context!!, R.color.black_transparent_50)))
        //        getWindow().addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
    }


    inner class FragmentLifeCycleHandler(fragment: Fragment) : Handler() {

        private val fragmentRef: WeakReference<Fragment>

        init {

            this.fragmentRef = WeakReference(fragment)
        }

        override fun dispatchMessage(msg: Message) {
            val fragment = fragmentRef.get() ?: return
            if (fragment.isAdded && fragment.activity != null) {
                super.dispatchMessage(msg)
            } else {
                Log.d(TAG, "Fragment under destroying")
            }
        }
    }
}